clang/lib/AST/ByteCode/InterpBuiltin.cpp

```diff
@@ -3523,6 +3523,21 @@ bool InterpretBuiltin(InterpState &S, CodePtr OpPC, const CallExpr *Call,
           return F;
         });
 
+  case X86::BI__builtin_ia32_vpmadd52luq128:
+  case X86::BI__builtin_ia32_vpmadd52luq256:
+  case X86::BI__builtin_ia32_vpmadd52luq512:
+    return interp__builtin_elementwise_triop(
+        S, OpPC, Call, [](const APSInt &A, const APSInt &B, const APSInt &C) {
+          return A + (B.trunc(52) * C.trunc(52)).trunc(52).zext(64);
```

Comment:
Outer trunc is unnecessary too.
```suggestion
          return A + (B.trunc(52) * C.trunc(52)).zext(64);
```